backend/apis/clients/list-clients.api.js

```diff
@@ -1,23 +1,61 @@
-const { app, databaseError, pool, invalidRequiest } = require("../../server");
+const { app, databaseError, pool, invalidRequest } = require("../../server");
 const mysql = require("mysql");
+const {
+  checkValid,
+  nullableValidInteger
+} = require("../utils/validation-utils");
 
 app.get("/api/clients", (req, res, next) => {
+  const validationErrors = checkValid(req.query, nullableValidInteger("page"));
+
+  if (validationErrors.length > 0) {
+    return invalidRequest(res.validationErrors);
+  }
+
   pool.getConnection((err, connection) => {
     if (err) {
       return databaseError(req, res, err);
     }
-    var queryString = "SELECT ? AS result";
 
-    const query = mysql.format(queryString, [req.query.foo]);
+    let queryString = `SELECT cl.id, cl.firstName, cl.lastName, cl.birthday, ct.zip, ct.primaryPhone, cl.addedBy as addedById,
+        us.firstName as addedByFirstName, us.lastname as addedByLastName, cl.dateAdded 
+      FROM 
+        clients cl 
+      JOIN 
+        contactInformation ct ON cl.id = ct.clientId 
```

Comment:
There is a bug here -- there can be multiple contactInformation rows for the same client. But we only want the most recent one. So we should select only the most recent row. See [this code](https://github.com/JustUtahCoders/comunidades-unidas-internal/blob/master/backend/apis/clients/get-client.api.js#L76) for an example of how to query only the most recent one.

Reply:
Got it... That make sense.